### ISSUE TITLE
Spell check preferences and togglable options

### ISSUE THREAD

**Issue by leagris** [raelgc/scudcloud#283]
I'd like to:
- Choose a default spell-checker in preferences.
- Mark a subset available spell-checkers for on-the-fly text entry selection.
- Have an icon aside text entry field to select spell-checker language (from subset above) or turn it on-off.
- Have the spell-checker automatically turned-off on typing ````some code````, `> quotes` and URI. Have it spell-check on anything else.

**Comment by raelgc** [comment]
Hi @leagris and sorry for my late response on this.

While all you mentioned will be great to have, there is a masterpiece component in the spell checker wrote in C code: the webkit spell checker plugin, which I borrowed from [QupZilla qtwebkit plugins project](https://github.com/QupZilla/qtwebkit-plugins/).

It automatically mark the misspelled words in red, based in which hunspell dictionary you have installed (we have no direct communication between ScudCloud and this plugin).

ScudCloud is working, in this case, suggesting the words in the context-menu and replacing them when user click on any.

So, basically, I need help in the C plugin code to finish this.

**Comment by aruetten** [comment]
What I did to switch language was editing ~/.local/share/applications/scudcloud.desktop

There is a line for the Exec, mine looks like:

`
Exec=env LANG=en_GB.UTF-8 /usr/bin/scudcloud
`

**Comment by raelgc** [comment]
Thanks for sharing this, @aruetten!

Additionally, if you're using any Debian based distro, remember to use `dpk-divert` to keep your changes between updates!

**Comment by leagris** [comment]
Thank you for sharing.
I think language switching should be user selectable and saved preference, at least per team and or per channel while interface can keep using system locale.
